test/e2e/metrics_test.go

```diff
@@ -89,134 +129,265 @@ func NewMetricsTestConfig(t *testing.T, client, selector, clusterRole, clusterBi
-	args = append(args, extraArgs...)
-	args = append(args, "--timeout=60s")
-	cmd := exec.CommandContext(ctx, client, args...)
-	output, err := cmd.CombinedOutput()
+// waitForPodDeletion polls until the named Pod no longer exists
+func waitForPodDeletion(ctx context.Context, t *testing.T, kubeClient kubernetes.Interface, namespace, name string) {
+	err := wait.PollUntilContextTimeout(ctx, 2*time.Second, 90*time.Second, false, func(ctx context.Context) (bool, error) {
+		pod, getErr := kubeClient.CoreV1().Pods(namespace).Get(ctx, name, metav1.GetOptions{})
+		if getErr != nil {
+			// The standard "not found" check
+			if strings.Contains(getErr.Error(), "not found") {
+				return true, nil
+			}
+			return false, getErr
+		}
+		// Some extra log info if the Pod is still around
+		t.Logf("Pod %q still present, phase=%q, deleting... (Timestamp=%v)",
+			name, pod.Status.Phase, pod.DeletionTimestamp)
+		return false, nil
+	})
 	if err != nil {
-		return fmt.Errorf("error waiting for deletion of %s %s: %v, output: %s", resourceType, resourceName, err, string(output))
+		if errors.Is(err, context.DeadlineExceeded) {
+			t.Fatalf("Timed out waiting for Pod %q to be deleted", name)
+		}
+		t.Logf("Error waiting for Pod %q deletion: %v", name, err)
+	} else {
+		t.Logf("Pod %q deleted", name)
 	}
-	return nil
 }
 
-// getComponentNamespace returns the namespace where operator-controller or catalogd is running
-func getComponentNamespace(t *testing.T, client, selector string) string {
-	cmd := exec.Command(client, "get", "pods", "--all-namespaces", "--selector="+selector, "--output=jsonpath={.items[0].metadata.namespace}")
-	output, err := cmd.CombinedOutput()
-	require.NoError(t, err, "Error determining namespace: %s", string(output))
+// getComponentNamespace identifies which Namespace is running a Pod that matches `selector`
+func getComponentNamespace(t *testing.T, kubeClient kubernetes.Interface, selector string) string {
+	t.Logf("Listing pods for selector %q to discover namespace", selector)
+	ctx := context.Background()
 
-	namespace := string(bytes.TrimSpace(output))
+	pods, err := kubeClient.CoreV1().Pods("").List(ctx, metav1.ListOptions{
+		LabelSelector: selector,
+	})
+	require.NoError(t, err, "Error listing pods for selector %q", selector)
+	require.NotEmpty(t, pods.Items, "No pods found for selector %q", selector)
+
+	namespace := pods.Items[0].Namespace
 	if namespace == "" {
-		t.Fatal("No namespace found for selector " + selector)
+		t.Fatalf("No namespace found for selector %q", selector)
 	}
 	return namespace
 }
 
-func stdoutAndCombined(cmd *exec.Cmd) ([]byte, []byte, error) {
-	var outOnly, outAndErr bytes.Buffer
-	allWriter := io.MultiWriter(&outOnly, &outAndErr)
-	cmd.Stdout = allWriter
-	cmd.Stderr = &outAndErr
-	err := cmd.Run()
-	return outOnly.Bytes(), outAndErr.Bytes(), err
+// Helpers for pointers
+func boolPtr(b bool) *bool {
+	return &b
+}
+
+func int64Ptr(i int64) *int64 {
+	return &i
```

Comment:
Could we remove it?
K8s provides the helper already.
The import is `"k8s.io/utils/ptr"`
And how to use it is:

```
RunAsNonRoot: ptr.To(true),

OR

RunAsUser:                ptr.To(int64(1001)),

```